setup.py

```diff
@@ -3,7 +3,8 @@
 import re
```

Comment:
@erikrose, I'd love a sanity check on the changes to this file if you get a chance.

Reply:
I don't see anything obviously wrong with it. I think it's more likely to help than hurt. Worst case, we missed some crazy combination of caches and Python versions, and we still have an old version of setuptools crash on encountering an env marker. But we should catch most of them this way.

Reply:
Thanks !